common/test/mock-ingester/src/lib.rs

```diff
@@ -25,6 +25,7 @@ use tokio_stream::wrappers::TcpListenerStream;
 const ROOT: &str = "/logs/agent";
 
 pub type FileLineCounter = Arc<Mutex<HashMap<String, FileInfo>>>;
+pub type ProcessFn = Arc<Mutex<Box<dyn Fn(&IngestBody) + Send>>>;
```

Comment:
This _really_ doesn't matter, but just for future reference because a Fn can be called as many times as you like it can't carry a reference to any state, so it _must_ be Sync implicitly (as there's no way 2 Fns can affect each other).

And since Arc is pretty much fat Box that carries an atomic usize around with it it means that this could just be Arc<dyn Fn(&IngestBody)>, no Mutex or 2nd indirection required.

Reply:
Good point, I don't remember why I added the `Mutex`... I'll remove it.